README.md
#DeepDark

A text-based browser game I've been working on since June 2016.

All game actions are performed by clicking one of the buttons below the text section or, in certain situations, clicking entries in the text section itself.

For debugging purposes, clicking any of the status variables will increment them, and shift clicking will decrement them.


changelog.txt
[6/24/16] {v0.0} Implemented main menu.
[6/26/16] {v0.1} Tweaked the posting function.
[6/26/16] {v0.1} Finalized the main menu.
[6/27/16] {v0.1} Created the options screen.
[6/27/16] {v0.2} Implemented death (starvation).
[6/27/16] {v0.2} Installed the notify function.
[6/28/16] {v0.2} Gave functionality to inventory items.
[6/28/16] {v0.3} Added sound (fire amibence).
[6/29/16] {v0.3} Added day and night ambience.
[6/30/16] {v0.3} Implemented time + day/night cycle.
[7/1/16] {v0.4} Added support for mobile browsers.
[7/1/16] {v0.5} Implemented crafting system (torch).

scripts/audio.js
var fading = false;

//$('.button').ready(function() {
	//$('.button').click(function() {
		var fireSound = new Audio('sound/fire.mp3');
		$(document).ready(function() {
			setInterval(function() {
				if (fire !== 0) {
					fireSound.play();
				} else {
					fireSound.pause();
				}
			}, 100);
			setInterval(function() {
				if (fire >= 0) {
					fireSound.volume = fire/100;
				} else {
					fireSound.volume = 0;
				}
			}, 100);
		}); //fire ambient sound
		/* function fireFadeIn() {
			console.log('Fading in the fire.');
			fading = true;
			fireSound.volume = 0;
			var fireFadeInterval = setInterval(function() {fireSound.volume += 0.01;console.log(fireSound.volume);}, 100);
			if (fireSound.volume >= 1) {
				clearInterval(fireFadeInterval);
			}
			fading = false;
		} */ //fire fade in [WIP]

		var nightSound = new Audio('sound/night.mp3');
		var daySound = new Audio('sound/birds.mp3');
		$(document).ready(function() {
			setInterval(function() {
				if (hour > 18 || hour < 6) {nightSound.play(); daySound.pause(); /*console.log("Time is night, playing night sound.");*/}
				else if (hour > 6 || hour < 18) {daySound.play(); nightSound.pause(); /*console.log("Time is day, playing day sound.");*/}
				else {console.log("The time is an unrecognized value.")}
			}, 1)
			setInterval(function() {
				if (fire > 0) {
					nightSound.volume = 0.05;
					daySound.volume = 0.05;
					}
				else {
					nightSound.volume = 0.1;
					daySound.volume = 0.1;
				}
			}, 100);
		}); //day/night ambient sound

		$(document).ready(function() {
			setInterval(function() {
				if (!soundOption) {
					fireSound.pause();
					nightSound.pause();
				}
			}, 1);
		}) //sound option toggle script
	//})
//})

scripts/cheats.js
var shifted = false;
$(document).ready(function() {
	$(document).keydown(function (e) {
		if (e.keyCode == 16) {
			shifted = true;
		}
	});
	$(document).keyup(function (e) {
		if (e.keyCode == 16) {
			shifted = false;
		}
	});
})

$(document).ready(function() {
	$(document).keyup(function(event) {
		if (event.keyCode == 74) {
			inventory.push('jerky');
			notify("Added jerky to inventory.");
		} //j key for jerky
		if (event.keyCode == 70) {
			inventory.push('firewood');
			notify("Added firewood to inventory.");
		} //f key for firewood
		if (event.keyCode == 84) {
			gameTick();
		} //t key for game tick
		if (event.keyCode == 187) {
			//console.log("It's " + time + ", adding 15 mins to be " + (time + 15) + "...");
			minute += 15;
		} //+ key for +30 mins
		if (event.keyCode == 189) {
			//console.log("It's " + time + ", subtracting 15 mins to be " + (time - 15) + "...");
			minute -= 15;
		} //- key for -15 mins
	});
	
	$(document).on('click', '.status', function() {
		//console.log('Clicked a status.');
		if ($(this).hasClass('fireStatus')) {
			if (shifted) {
				fire -= 10;
			} else {
				fire += 10;
			}
		} else if ($(this).hasClass('tempStatus')) {
			if (shifted) {
				temp -= 10;
			} else {
				temp += 10;
			}
		} else if ($(this).hasClass('healthStatus')) {
			if (shifted) {
				health -= 10;
			} else {
				health += 10;
			}
		} else if ($(this).hasClass('hungerStatus')) {
			if (shifted) {
				hunger -= 10;
			} else {
				hunger += 10;
			}
		} else if ($(this).hasClass('timeStatus')) {
			if (shifted) {
				minute -= 15;
			} else {
				minute += 15;
			}
		} else if ($(this).hasClass('dateStatus')) {
			if (shifted) {
				day -= 1;
			} else {
				day += 1;
			}
		} else if ($(this).hasClass('timeStatus')) {
			if (shifted) {
				time -= 30;
			} else {
				time += 30;
			}
		}
	});
})

scripts/engine.js
function gameTick() {
	timeTick();
	fireTick();
	tempTick();
	healthTick();
	hungerTick();
	randomTick();
	saveGame();
} //ticks the game engine

var timeIndicator = "";
function timeTick() {
	minute += 15;
	if (hour === 18) {
		time = 'night';
		timeIndicator = "The sun is setting.";
	} else if (hour === 6) {
		time = 'day';
		timeIndicator = "The sun is rising.";
	} else if (hour >= 19 || hour <= 5) {
		time = 'night';
		timeIndicator = "It's night time.";
	} else if (hour >= 7 || hour <= 17) {
		time = 'day';
		timeIndicator = "It's daytime.";
	} else {
		time = 'apocalypse';
		timeIndicator = "Time exists only in human perception.";
	}
	overwrite(timeIndicator, '.timeIndicator');
} //sets time var and time indicator
$(document).ready(function() {
	setInterval(function() {
		/* if (time < 1000) {
			if (time < 100) {
				if (time > 0) {
					timeString = "00" + time.toString();
				} else {
					timeString = "0000";
				}
			} else {
				timeString = "0" + time.toString();
			}
		} else {
			timeString = time.toString();
		} */ //converts the time into HH:MM format}, 1);
		var hourString = hour.toString();
		var minString = minute.toString();
		var amPM = '';
		/* if (hour < 10) {
			hourString = '0' + hour.toString();
		} else {hourString = hour.toString();} */
		if (hour === 0) {
			hourString = '12';
			amPM = 'AM';
		} else if (hour > 0 && hour < 12) {
			hourString = hour.toString();
			amPM = amPM = 'AM';
		} else if (hour == 12) {
			hourString = '12';
			amPM = 'PM';
		} else {
			hourString = (hour - 12).toString();
			amPM = 'PM';
		}
		if (minute < 10) {
			minString = '0' + minute.toString();
		} else {minString = minute.toString();}
		timeString = hourString + ':' + minString + ' ' + amPM;
	}, 1);
}); //sets the timeString

setInterval(function() {
	if (month === 1 ||
		month === 3 ||
		month === 5 ||
		month === 7 ||
		month === 8 ||
		month === 10) {
		if (day == 32) {
			month++;
			day = 1;
		}
	} else if (month === 4 ||
			   month === 6 ||
			   month === 9 ||
			   month === 11) {
		if (day === 31) {
			month++;
			day = 1;
		}
	} else if (month === 2) {
		if (year % 4 === 0) {
			if (day === 30) {
				month++;
				day = 1;
			}
		} else if (day == 29) {
			month++;
			day = 1;
		}
	} else if (month === 12) {
		if (day === 32) {
			year++;
			month = 1;
			day = 1;
		}
	}
}, 1); //advances the calendar by one day
$(document).ready(function() {
	setInterval(function() {
		dateString = month + "/" + day + "/" + year
	}, 1);
}); //sets the dateString

var fireIndicator = "";
function fireTick() {
	if ((fire > 0)) {fire -= Math.round(Math.random() * 6);} else {fire = 0;}
	if (fire > 75) {
		fireIndicator = 'The fire is roaring.';
	} else if (fire > 50) {
		fireIndicator = 'The fire is going strong.';
	} else if (fire > 25) {
		fireIndicator = 'The fire is starting to fade.';
	} else if (fire > 10) {
		fireIndicator = 'The fire is dwindling.';
	} else if (fire > 0) {
		fireIndicator = 'The fire is almost extinguished.';
	} else {
		$('.button:contains("fuel fire")').text('light fire');
		fireIndicator = 'The fire has gone out.';
	}
	overwrite(fireIndicator, '.fireIndicator');
} //ticks the fire var
function fuelFire() {
	if (fire != 100) {
		if (xContainsY(inventory, 'firewood')) {
			var fireGrowBy = 0;
			while (fireGrowBy < 14 || fireGrowBy > 17) {
				var fireGrowBy = Math.round(Math.random() * 30);
			}
			//console.log("Growing fire by " + fireGrowBy + "...");
			fire += fireGrowBy;
			if (fire > 100) {fire = 100;}
			removeFromBag('firewood');
			notify("The fire grows by " + fireGrowBy + "%.");
		} else {
			notify("There's no more firewood.");
		}
	} else {
		notify("The fire is big enough already.");
	}
	saveGame();
} //throws a log on the fire
function lightFire() {
	if (xContainsY(inventory, 'firewood')) {
		if (xContainsY(inventory, 'lighter')) {
			fuelFire();
			notify("Gotta use the lighter and some firewood.");
			$('.button:contains("light fire")').text('fuel fire');
		} else {
			notify("You need the lighter.");
		}
	} else {
		notify("There's no more firewood.");
	}
	saveGame();
} //lights the fire if it's gone out

var tempIndicator = "";
function tempTick() {
	if (place == 'home' && fire > 0) {
		if (fire > 75) {
			temp += 10;
		} else if (fire > 50) {
			temp += 8;
		} else if (fire > 25) {
			temp += 6;
		} else if (fire > 10) {
			temp += 4;
		} else if (fire > 0) {
			temp += 2;
		}
	} else {
		if (temp > 0) {
			if (time === 'night') {
				temp -= 5;
			} else {
				temp -= 3;
			}
		} else {
			temp = 0;
		}
	}
	if (temp > 75) {
		tempIndicator = "You are warm.";
	} else if (temp > 50) {
		tempIndicator = "You're getting colder.";
	} else if (temp > 25) {
		tempIndicator = "You are cold.";
	} else if (temp > 10) {
		tempIndicator = "You are very cold.";
	} else if (temp > 0) {
		tempIndicator = "You are extremely cold.";
	} else {
		tempIndicator = 'You are freezing. Get to the fire.';
		if (health > 0) {
			health -= 3;
		} else {
			death('hypothermia');
		}
	}
	if (temp > 100) {
		temp = 100;
	}
	overwrite(tempIndicator, '.tempIndicator');
} //ticks the temp var

var healthIndicator = "";
function healthTick() {
	if (health > 75) {
		healthIndicator = "You are healthy.";
	} else if (health > 50) {
		healthIndicator = "You are injured.";
	} else if (health > 25) {
		healthIndicator = "You need medical attention.";
	} else if (health > 10) {
		healthIndicator = "You are in critical condition.";
	} else if (health > 0) {
		healthIndicator = "You are about to die.";
	} else {
		healthIndicator = 'You have died.';
	}
	if (place === 'home' && fire > 0 && temp > 25 && hunger > 25) {
		health += 10;
	}
	overwrite(healthIndicator, '.healthIndicator');
}

var hungerMultiplier = 1;
var hungerIndicator = "";
function hungerTick() {
	if (hunger > 0) {hunger -= Math.round(Math.random() * hungerMultiplier);} else {hunger = 0;}
	//hunger -= Math.round(Math.random() * hungerMultiplier);
	if (hunger > 75) {
		hungerIndicator = 'You are full.';
	} else if (hunger > 50) {
		hungerIndicator = 'You are hungry.';
	} else if (hunger > 25) {
		hungerIndicator = 'You are very hungry.';
	} else if (hunger > 10) {
		hungerIndicator = 'You are extremely hungry.';
	} else if (hunger > 0) {
		fireIndicator = 'You are on the brink of starvation.';
	} else {
		hungerIndicator = 'You are starving. Eat something.';
		if (health > 0) {
			health -= 5;
		} else {
			death('starvation');
		}
	}
	overwrite(hungerIndicator, '.hungerIndicator');
} //ticks the hunger var

function randomTick() {
	if (Math.round(Math.random() * 100) > 95) {
	}
} //chance of triggering a special notify [WIP]

$(document).ready(function() {
	setInterval(function() {
		if (fire > 100) {
			fire = 100;
		} else if (fire < 0) {
			fire = 0;
		}
		if (temp > 100) {
			temp = 100;
		} else if (temp < 0) {
			temp = 0;
		}
		if (health > 100) {
			health = 100;
		} else if (health < 0) {
			health = 0;
		}
		if (hunger > 100) {
			hunger = 100;
		} else if (hunger < 0) {
			hunger = 0;
		}
		if (hour > 23) {
			hour = 0;
			day++;
		}
		if (minute >= 60) {
			hour += 1;
			minute = 0;
		}
		if (minute <= -1) {
			hour -= 1;
			minute = 45;
		}
	}, 1);
}); //keep all stats above min and below max

function saveGame() {
	/*var invString = '';
	for (i=0;i<=inventory.length;i++) {
		invString += ',' + inventory[i];
	} */
	document.cookie = "inventory=" + inventory;
	//console.log("inventory cookie: " + getCookie('inventory'));
	document.cookie = "fire= " + fire;
	document.cookie = "temp= " + temp;
	document.cookie = "health= " + health;
	document.cookie = "hunger= " + hunger;
	document.cookie = "hour= " + hour;
	document.cookie = "minute= " + minute;
	document.cookie = "month= " + month;
	document.cookie = "day= " + day;
	document.cookie = "year= " + year;
	document.cookie = "place= " + place;
} //saves the game using cookies

function getCookie(cname) {
    var name = cname + "=";
    var ca = document.cookie.split(';');
    for(var i = 0; i <ca.length; i++) {
        var c = ca[i];
        while (c.charAt(0)==' ') {
            c = c.substring(1);
        }
        if (c.indexOf(name) == 0) {
            return c.substring(name.length,c.length);
        }
    }
    return "";
} //gets the cookie value of the specified cookie name

scripts/interface.js
function post(input) {
	$('#posts').append("<p class='fadein'>" + input + "</p>");
    $('.fadein').hide().fadeIn('slow').removeClass('fadein');
    var element = document.getElementById("posts");
    element.scrollTop = element.scrollHeight;
} //creates a new post
function postWithClass(input, postWith) {
	$('#posts').append("<p class='fadein'>" + input + "</p>");
    $('.fadein').hide().fadeIn('slow').addClass(postWith).removeClass('fadein');
    var element = document.getElementById("posts");
    element.scrollTop = element.scrollHeight;
} //posts with a class
function overwrite(input, classToOverwrite) {
	//console.log("Overwriting class '" + classToOverwrite + "' with message '" + input + "'...");
	$(classToOverwrite).text(input);
} //overwrites a class
function clearPosts() {
	$('#posts').empty();
} //deletes all posts
function fadePosts () {
	$('p').fadeOut('fast');
} //fades out all posts

var numOfButtons = 0;
function newButton(buttonText) {
	numOfButtons += 1;
	$('#buttons').append("<div class='button button" + numOfButtons + "'>" + buttonText + "</div>");
	$('.button').width("calc(" + 100 / numOfButtons + "% - 8px");
} //creates one new button
function clearButtons() {
	$('#buttons').empty();
	numOfButtons = 0;
} //deletes all buttons
function fadeButtons() {
	$('.button').fadeOut('fast');
} //fades out all buttons
function triButtons(text1, text2, text3) {
	clearButtons();
	newButton(text1);
	newButton(text2);
	newButton(text3);
} //creates three new buttons
function newButtons(buttonTexts) {
	clearButtons();
	console.log("buttonTexts length is " + buttonTexts.length);
	for (var i=0;i==buttonTexts.length;i++) {
		console.log("Making button from buttonTexts[" + i + "]...");
		newButton(buttonTexts[i]);
	}
} //creates buttons on a case-by-case basis

//uses document because document will be topmost level in bubbling
/*$(document).on('touchmove',function(e){
  e.preventDefault();
});
//uses body because jquery on events are called off of the element they are
//added to, so bubbling would not work if we used document instead.
$('body').on('touchstart','.scrollable',function(e) {
  if (e.currentTarget.scrollTop === 0) {
    e.currentTarget.scrollTop = 1;
  } else if (e.currentTarget.scrollHeight === e.currentTarget.scrollTop + e.currentTarget.offsetHeight) {
    e.currentTarget.scrollTop -= 1;
  }
});
//prevents preventDefault from being called on document if it sees a scrollable div
$('body').on('touchmove','.scrollable',function(e) {
  e.stopPropagation();
}); */

var notifTimeout = setTimeout(function(){}, 0);
function notify(input, seconds) {
	clearTimeout(notifTimeout);
	$('.notification').hide().text(input).fadeIn('slow');
	//console.log(seconds);
	if (seconds == undefined) {
		notifTimeout = setTimeout(function() {
			$('.notification').fadeOut('slow');
		}, 2500);
	} else {
		notifTimeout = setTimeout(function() {
			$('.notification').fadeOut('slow');
		}, seconds);
	}
} //shows a notif at the bottom of the screen

function xContainsY(array, value) {
	return(array.indexOf(value) > -1);
} //check if an array contains a value
function findXInY(value, array) {
	return(array.indexOf(value));
}

dying = false;
function mainMenu() {
	clearPosts();
	clearButtons();
	//console.log("Initializing main menu...");
	triButtons("load game", "new game", "options");
	if (!dying) {
		splash();
	} else {
		dying = false;
	}
} //initialize main menu
function splash() {
	//post("</p><br/><p>It's getting dark.</p><br/><p>Keep the fire going.</p><br/><p>Oh, and try not to starve.");
	post("The fire is going out.");
	post("Gotta keep the fire going.");
	//for (i=0;i<=10;i++) {
	//	post(Math.random());
	//}
} //posts the changelog [DEFUNCT]

function statusUpdate() {
	$('.fireStatus').text('fire: ' + fire + '%');
	$('.tempStatus').text('temp: ' + temp + '%');
	$('.healthStatus').text('health: ' + health + '%');
	$('.hungerStatus').text('hunger: ' + hunger + '%');
	$('.timeStatus').text(timeString);
	$('.dateStatus').text(dateString);
	/* var countdown = 0
	for (i=1;i<6;i++) {
		$('.statustitle').text('status (' + countdown + 's)');
		setTimeout(function() {countdown += 1}, 1000);
	} */
} //updates the info in the status box
$(document).ready(function() {
	setInterval(function() {statusUpdate();}, 50);
}); //loops the status function every 0.1 secs

var statusOption = true;
var invertColorsOption = false;
var soundOption = true;
function showOptions() {
	clearPosts();
	clearButtons();
	newButton("main menu");
	//postWithClass('show status: ' + statusOption, 'statusOption');
	postWithClass('invert colors: ' + invertColorsOption, 'invertColorsOption');
	postWithClass('sound: ' + /*function() {
		if (soundOption) {
			return 'on';
		} else {
			return 'off';
		}
	}*/ soundOption, 'soundOption');
	//postWithClass('simulate death (debug)', 'simDeath');
} //creates the options screen
$(document).ready(function() {
	$('#posts').on('click', '.statusOption', function() {
		if (statusOption) {
			statusOption = false;
			$('#status').css('visibility', 'hidden');
			overwrite('status: ' + statusOption, '.statusOption');
		} else {
			statusOption = true;
			$('#status').css('visibility', 'visible');
			overwrite('status: ' + statusOption, '.statusOption');
		}
	}); //toggle status
	$('#posts').on('click', '.invertColorsOption', function() {
		if (invertColorsOption) {
			invertColorsOption = false;
			$('body').removeClass('inverted');
			overwrite('invert colors: ' + invertColorsOption, '.invertColorsOption');
		} else {
			invertColorsOption = true;
			$('body').addClass('inverted');
			overwrite('invert colors: ' + invertColorsOption, '.invertColorsOption');
		}
	}); //invert colors
	$('posts').on('click', '.soundOption', function() {
		if (soundOption) {
			soundOption = false;
			overwrite('sound: ' + soundOption, '.soundOption');
		} else {
			soundOption = true;
			overwrite('sound: ' + soundOption, '.soundOption');
		}
	});
	$('#posts').on('click', '.simDeath', function(){death('debug');}); //fake die
}); //gives function to options posts

function pickRandomFrom(list) {
	return list[Math.floor(Math.random()*list.length)];
}

function timeShift(time) {
	if (time === 'day') {
		$('body').css('background-color', '#87ceeb');
		/* $('#container').css('background-color', 'white');
		$('#container').css('border', '1px solid black');
		$('*').css('color', 'black'); */
	}
	if (time === 'sunset') {
		$('body').css('background-color', '#fe4b21');
	}
	if (time === 'night') {
		$('body').css('background-color', 'black');
	}
} //shifts background-color for time of day

function death(causeOfDeath) {
	dying = true;
	clearInterval(gameInterval);
	mainMenu();
	clearPosts();
	post('<br/>');
	postWithClass('You have died of ' + causeOfDeath + '.', 'deathMessage');
	setTimeout(function() {postWithClass('Score: ' + score, 'score');}, 250);
	setTimeout(function() {post('<br/><br/>');}, 500);
	var quotes = ['That which does not kill us makes us stronger.', 'When you look into an abyss, the abyss also looks into you.', 'There are no facts, only interpretations.', 'Convictions are more dangerous enemies of truth than lies.', 'There is always some madness in love. But there is also always some reason in madness.', 'He who has a why to live can bear almost any how.', 'All truly great thoughts are conceived by walking.', 'To live is to suffer, to survive is to find some meaning in the suffering.', 'We love life, not because we are used to living but because we are used to loving.', 'Without music, life would be a mistake.', 'Thoughts are the shadows of our feelings - always darker, emptier and simpler.'];
	setTimeout(function() {post(pickRandomFrom(quotes));}, 1500);
	setTimeout(function() {post('-Friedrich Nietzsche');}, 2000);
} //message that shows on death

time = 'night';
timeString = 'unset';
dateString = 'unset';
hour = 0;
minute = 0;
month = 0;
day = 0;
year = 0;
var fire = 0;
var temp = 0;
var health = 0;
var hunger = 0;
var score = 0;
var place = 'main menu';

scripts/game.js
var gameInterval = undefined;
function newGame() {
	inventory = ['flashlight', 'lighter', 'jerky', 'firewood', 'firewood'];
	hour = 23;
	minute = 15;
	month = 11;
	day = 25;
	year = 15;
	fire = 75;
	temp = 100;
	health = 87;
	hunger = 90;
	score = 0;
	goHome();
	gameInterval = setInterval(function() {gameTick();}, 5000);
} //creates a new game
function loadGame() {
	//alert('Saving and loading has not yet been implemented.');
	var tempInv = getCookie('inventory');
	var tempInvString = tempInv.toString();
	//inventory = JSON.parse(JSON.stringify(getCookie('inventory')));
	inventory = getCookie('inventory').split(',');
	console.log("parsed inventory: " + inventory);
	fire = getCookie('fire');
	temp = getCookie('temp');
	health = getCookie('health');
	hunger = getCookie('hunger');
	hour = getCookie('hour');
	minute = getCookie('minute');
	month = getCookie('month');
	day = getCookie('day');
	year = getCookie('year');
	place = getCookie('place');
	if (place === 'home') {
		goHome();
	}
	setTimeout(function() {gameInterval = setInterval(function() {gameTick();}, 5000);}, 5000);
} //loads an existing save [WIP]

$(document).ready(function() {
	mainMenu();
	$('#buttons').on('click', '.button', function() {
		switch($(this).text()) {
			case 'new game':
				post("...");
				setTimeout(function() {post("Gotta find her.");}, 100);
				fadeButtons();
				setTimeout(function() {newGame();}, 300);
				break;
			case 'load game':
				/* fadePosts();
				post("Welcome home.");
				fadeButtons(); */
				setTimeout(loadGame(), 500);
				break;
			case 'options':
				showOptions();
				break;
			case 'main menu':
				mainMenu();
				break;
			case 'inventory':
				saveGame();
				openBag();
				break;
			case 'close bag':
				saveGame();
				closeBag();
				break;
			case 'craft':
				craft();
				break;
			case 'fuel fire':
				fuelFire();
				break;
			case 'light fire':
				lightFire();
				break;
			case 'embark':
				saveGame();
				embark();
				break;
			case 'return':
				saveGame();
				goHome();
				break;
			case 'forage':
				saveGame();
				forage();
				break;
			case 'adventure':
				saveGame();
				notify("I'm not ready to go yet.");
				break;
		}
	});
}); //gives function to buttons

function goHome() {
	place = 'home';
	fadePosts();
	triButtons('inventory', 'fuel fire', 'embark');
	postWithClass('Looking at the clock...', 'timeIndicator');
	postWithClass('Inspecting the fire..', 'fireIndicator');
	postWithClass('Taking your temperature...', 'tempIndicator');
	postWithClass('Checking your stomach...', 'hungerIndicator');
	postWithClass('Administering a physical...', 'healthIndicator');
	gameTick();
} //shows the home screen with friendly stats

function openBag() {
	postsBackup = $('#posts').html();
	buttonsBackup = $('#buttons').html();
	clearPosts(); clearButtons();
	newButton('close bag');
	newButton('craft');
	refreshBag();
	$('#posts').addClass('bag');
} //opens the inventory screen
function closeBag() {
	clearPosts(); clearButtons();
	$('#posts').html(postsBackup);
	$('#buttons').html(buttonsBackup);
	$('#posts').removeClass('bag');
} //closes the inventory screen
function refreshBag() {
	clearPosts(); clearButtons();
	newButton('close bag');
	newButton('craft');
	var firstOfThis = function() {
		var matches = [];
		for (j=0;j<=inventory.length;j++) {
			if (currentItem === inventory[j]) {
				matches.push(j);
			}
		}
		return(matches[0]);
	};
	var i = 0;
	for (i=0;i<=inventory.length-1;i++) {
		var currentItem = inventory[i];
		//console.log("Dealing with item " + i + ", which is a " + currentItem + "...");
		//console.log("First " + currentItem + " is at spot " + firstOfThis() + ".");
		//console.log("Current " + currentItem + " is at spot " + i + ".");
		if (count(currentItem) > 1) {
			//console.log(currentItem + " count is greater than 1.");
			if (i == firstOfThis()) {
				//console.log("This is the first " + currentItem + ", posting with an amount.");
				$('#posts').append('<p class="bagItem stacked" id="' + currentItem + '">' + currentItem + ' (' + count(currentItem) + ')</p>');
			} else {/*console.log(console.log("This isn't the first " + currentItem + ", not posting."))*;*/}
		} else {
			//console.log("This is the only " + currentItem + ", posting with no amount.");
			$('#posts').append('<p class="bagItem" id="' + currentItem + '">' + currentItem + '</p>');
		}
		//console.log("Done with item " + i + ", moving on to item " + (i+1) + "...");
	}
	$('#posts').addClass('bag');
} //refreshes bag for after crafting

function addToBag(item) {
	inventory.push(item);
	notify("You obtain " + item + ".");
} //add an item to the inventory
permanentItems = ['flashlight', 'lighter']; //items that can't be removed
function removeFromBag(item) {
	if (!(xContainsY(permanentItems, item))) {
		for (i=inventory.length;i>=0;i--) {
			if (inventory[i] == item) {
				inventory.splice(i, 1);
				break;
			}
		}
	}
} //removes an item from the inventory

function count(item) {
	var number = 0;
	if (item === 'food') {
		for (i=0;i<inventory.length;i++) {
			if (xContainsY(food, inventory[i])) {
				number++;
			}
		}
	} else {
		for (i=0;i<inventory.length;i++) {
			if (inventory[i] === item) {
				number++;
			}
		}
	}
	return(number);
} //count the number of something in the bag

$(document).ready(function() {
	$('#posts').on('click', '.bagItem', function() {
		if (crafting) {
			$(this).fadeOut('fast');
		} else {
			var $this = $(this);
			var $thisText = $(this).text();
			var activeItem = 'unset';
			if ($this.hasClass('stacked')) {
				//console.log("activeItem: " + $thisText.substring(0, $thisText.length - 4));
				activeItem = $thisText.substring(0, $thisText.length - 4);
			} else {
				//console.log("activeItem: " + $thisText);
				activeItem = $thisText;
			}
			$(this).addClass('activeItem');
			if (xContainsY(food, activeItem)) {
				$(this).addClass('eating');
				eat(findXInY(activeItem, food), activeItem);
			} else {
				switch(activeItem) {
					case 'flashlight':
						notify("I can use this to go out at night.");
						break;
					case 'lighter':
						notify("I'll use this if the fire goes out.");
						break;
					case 'firewood':
						notify("Gotta keep the fire going.");
						break;
					case 'ashes':
						notify("Not much use for these.");
						break;
				};
			};
		}
	});
}); //interaction with the inventory screen

var food = ['jerky', 10, 'granola', 15, 'red berry', 5, 'blueberry', 5, "green berry", 5, 'pink berry', 5, 'strawberry', 5]; //list of foods and hunger values
var berries = ['red berry', 'blueberry', "green berry", "pink berry", "strawberry"]
function eat(foodIndex, foodName) {
	var foodHungerVal = food[foodIndex + 1];
	//console.log('Attempting to eat ' + foodName + ' with hunger value ' + foodHungerVal + '...')
	if ((hunger + foodHungerVal) <= 100) {
		//console.log('Success, eating now.');
		removeFromBag(foodName);
		refreshBag();
		hunger += foodHungerVal;
		notify('The ' + foodName + ' replenishes ' + foodHungerVal + ' hunger.');
	} else {notify("I'm not hungry enough to eat this.");};
}; //eat something by index in food array

var crafting = false;
var recipes = [['firewood', 'lighter', 'torch'], ['jerky', 'lighter', 'ashes']]; //0=item1, 1=item2, 3=given result
function craft() {
	crafting = true;
	//console.log("Initiating crafting...")
	clearButtons();
	newButton('cancel');
	$(document).on('click', '.button:contains("cancel")', function() {
		console.log("Aborting craft.");
		refreshBag();
		return;
	});
	var item1 = '';
	var item2 = '';
	//console.log("Waiting for item inputs.")
	$("#posts").on('click', 'p', function() {
		var $this = $(this);
		var $thisText = $(this).text();      
		if (item1.length === 0) {
			if ($this.hasClass('stacked')) {
				//console.log("item1: " + $thisText.substring(0, $thisText.length - 4));
				item1 = $thisText.substring(0, $thisText.length - 4);
			} else {
				//console.log("item1: " + $thisText);
				item1 = $thisText;
			}
		} else if (item2.length === 0) {
			if ($this.hasClass('stacked')) {
				//console.log("item2: " + $thisText.substring(0, $thisText.length - 4));
				item2 = $thisText.substring(0, $thisText.length - 4);
			} else {
				//console.log("item2: " + $thisText);
				item2 = $thisText;
			}
		} else {
			return;
		}
	})
	var readyToCraft = false;
	var craftInterval = setInterval(function() {
		if (item1.length > 0 && item2.length > 0) {
			readyToCraft = true;
			actuallyCraft();
		}
	})
	function actuallyCraft() {
		clearInterval(craftInterval);
		//console.log("Checking craftability...")
		var canCraft = false;
		var craftResult = '';
		for (i=0;i<recipes.length;i++) {
			for (j=0;j<=1;j++) {
				if (recipes[i][j] === item1) {
					if (j === 0) {
						if (recipes[i][1] === item2) {
							canCraft = true;
							craftResult = recipes[i][2];
						}
					} else {
						if (recipes[i][0] === item2) {
							canCraft = true;
							craftResult = recipes[i][2];
						}
					}
				} else if (recipes[i][j] === item2) {
					if (j === 0) {
						if (recipes[i][1] === item1) {
							canCraft = true;
							craftResult = recipes[i][2];
						}
					} else {
						if (recipes[i][0] === item1) {
							canCraft = true;
							craftResult = recipes[i][2];
						}
					}
				}
			}
		}
		if (canCraft) {
			//console.log("Success, crafting " + craftResult + ".");
			removeFromBag(item1);
			removeFromBag(item2);
			addToBag(craftResult);
			notify(item1 + " + " + item2 + " = " + craftResult, 4000);
		} else {
			//console.log("Failed, not crafting.")
			notify("I can't make anything with these.", 4000);
		}
		setTimeout(function() {refreshBag();}, 500);
		crafting = false;
		return;
	}
} //craft two items into a new item

function embark() {
	place = "home";
	fadePosts();
	triButtons('return', 'forage', 'adventure');
	postWithClass("Firewood: " + count('firewood'), 'firewoodCount');
	postWithClass("Food: " + count('food'), 'foodCount');
} //shows the embark screen

function forage() {
	var woodChance = Math.random();
	var numOfWood = 0;
	var berryChance = Math.random();
	var numOfBerries = 0;
	var timeItTook = Math.random();
	if (woodChance > 0.9) {
		addToBag('firewood');
		addToBag('firewood');
		addToBag('firewood');
		numOfWood = 3;
	} else if (woodChance > 0.7) {
		addToBag('firewood');
		addToBag('firewood');
		numOfWood = 2;
	} else if (woodChance > 0.5) {
		addToBag('firewood');
		numOfWood = 1;
	} else {
		numOfWood = 0;
	}
	if (berryChance > 0.9) {
		addToBag(pickRandomFrom(berries));
		addToBag(pickRandomFrom(berries));
		addToBag(pickRandomFrom(berries));
		numOfBerries = 3;
	} else if (berryChance > 0.7) {
		addToBag(pickRandomFrom(berries));
		addToBag(pickRandomFrom(berries));
		numOfBerries = 2;
	} else if (berryChance > 0.5) {
		addToBag(pickRandomFrom(berries));
		numOfBerries = 1;
	} else {
		numOfBerries = 0;
	}
	if (timeItTook > 0.75) {
		gameTick();
		timeItTook = 15;
	} else if (timeItTook > 0.5) {
		gameTick();
		gameTick();
		timeItTook = 30;
	} else {
		gameTick();
		gameTick();
		gameTick();
		timeItTook = 45;
	}
	notify("Firewood: +" + numOfWood + " | Berries: +" + numOfBerries + " | Time: " + timeItTook + " mins", 6000);
	overwrite("Firewood: " + count('firewood'), '.firewoodCount');
	overwrite("Food: " + count('food'), '.foodCount');
} //gathers berries and firewood